### PR TITLE
feat(plugins): enable ai plugins to read request body from buffered file

### PR DESCRIPTION
**Pull request by fffonion**
### Summary

- [feat(plugins): enable ai plugins to read request body from buffered file](https://github.com/Kong/kong/commit/f2eb35eeb0991f5e9a2fd63f8ede5fb9d4b03f49)
- [feat(pdk): add argument to allow request body read from disk](https://github.com/Kong/kong/commit/2771163eacc01276fdcc6f6f8159313b578c4052)

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

AG-39

#13211

### CODE REVIEW

**Review by liverpool8056 — COMMENTED** [review]
Do we need update `removed_fields` correspondingly?

**Comment by fffonion** [comment]
@liverpool8056 yes, i didn't notice removed_fields is also present on CE, will fix.

**Review by liverpool8056 — APPROVED** [review]

**Review by jschmid1 — APPROVED** [review]

**Comment by fffonion** [comment]
I did a force push to port the backward compat tests from EE PR.

**Comment by locao** [comment]
@fffonion could you take care of the manual cherry-pick, please?

File: kong/pdk/request.lua [hunk 1]
```diff
@@ -693,30 +694,69 @@ local function new(self)
...
+        return ""
+      end
+
+      if not max_allowed_file_size then
         return nil, "request body did not fit into client body buffer, consider raising 'client_body_buffer_size'"
+      end
 
-      else
-        return ""
+      local file, err = io.open(body_file, "r")
```
Comment (dol) [inline kong/pdk/request.lua:728]: IMHO it's safer to read the file content as binary. `local file, err = io.open(body_file, "rb")`. Just to be safe. https://www.lua.org/pil/21.2.2.html

**Pull request merged**